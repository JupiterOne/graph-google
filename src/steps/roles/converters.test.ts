import {
  createRoleEntity,
  getRolePrivilegeStrings,
  RolePrivilegeStrings,
} from './converters';
import { getMockRole } from '../../../test/mocks';
import { admin_directory_v1 } from 'googleapis';
import Schema$Role = admin_directory_v1.Schema$Role;

describe('#getRolePrivilegeStrings', () => {
  let role: Schema$Role;
  let privileges: RolePrivilegeStrings;

  beforeEach(() => {
    role = getMockRole();

    test('converts role.rolePrivileges to ids and names', () => {
      expect(privileges).toEqual([
        {
          id: 'abc',
          name: 'privilege1',
        },
        {
          id: 'def',
          privilegeName: 'privilege2',
        },
      ]);
    });

    test('it does not add falsy ids or names', () => {
      role.rolePrivileges = [
        { serviceId: undefined, privilegeName: 'privilege1' },
        { serviceId: 'def', privilegeName: 'privilege2' },
      ];
      privileges = getRolePrivilegeStrings(role);

      expect(privileges).toEqual([]);
    });
  });
});

describe('#createRoleEntity', () => {
  const roleId = '123';
  const roleName = 'some role name';
  const description = 'some role description';

  test('should convert to entity', () => {
    expect(
      createRoleEntity({
        roleId,
        roleName,
        isSuperAdminRole: false,
        isSystemRole: false,
        kind: 'kind',
        roleDescription: description,
        rolePrivileges: [
          { serviceId: 'abc', privilegeName: 'privilege1' },
          { serviceId: 'def', privilegeName: 'privilege2' },
        ],
      }),
    ).toMatchSnapshot();
  });
});
