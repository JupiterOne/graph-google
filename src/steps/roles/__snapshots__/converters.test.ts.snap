// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createRoleEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "AccessRole",
  ],
  "_key": "google_role_123",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "isSuperAdminRole": false,
        "isSystemRole": false,
        "kind": "kind",
        "roleDescription": "some role description",
        "roleId": "123",
        "roleName": "some role name",
        "rolePrivileges": Array [
          Object {
            "privilegeName": "privilege1",
            "serviceId": "abc",
          },
          Object {
            "privilegeName": "privilege2",
            "serviceId": "def",
          },
        ],
      },
    },
  ],
  "_type": "google_role",
  "createdOn": undefined,
  "description": "some role description",
  "displayName": "some role name",
  "id": "123",
  "isAdmin": false,
  "isSystem": false,
  "kind": "kind",
  "name": "some role name",
  "privileges": Array [
    Object {
      "id": "abc",
      "name": "privilege1",
    },
    Object {
      "id": "def",
      "name": "privilege2",
    },
  ],
  "vendor": "Google",
}
`;
